code/modules/mob/living/basic/space_fauna/morph.dm

```diff
@@ -0,0 +1,215 @@
+	habitable_atmos = list("min_oxy" = 0, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minimum_survivable_temperature = TCMB
+
+	obj_damage = 50
+	melee_damage_lower = 20
+	melee_damage_upper = 20
+
+	// Oh you KNOW it's gonna be real green
+	lighting_cutoff_red = 10
+	lighting_cutoff_green = 35
+	lighting_cutoff_blue = 15
+
+	attack_verb_continuous = "glomps"
+	attack_verb_simple = "glomp"
+	attack_sound = 'sound/effects/blobattack.ogg'
+	attack_vis_effect = ATTACK_EFFECT_BITE //nom nom nom
+	butcher_results = list(/obj/item/food/meat/slab = 2)
+
+	ai_controller = /datum/ai_controller/basic_controller/morph
+
+	/// A weakref pointing to the form we are currently assumed as.
+	var/datum/weakref/form_weakref = null
+	/// A typepath pointing of the form we are currently assumed as. Remember, TYPEPATH!!!
+	var/atom/movable/form_typepath = null
+	/// The ability that allows us to disguise ourselves.
+	var/datum/action/cooldown/mob_cooldown/assume_form/disguise_ability = null
+
+	/// How much damage are we doing while disguised?
+	var/melee_damage_disguised = 0
+	/// Can we eat while disguised?
+	var/eat_while_disguised = FALSE
+
+/mob/living/basic/morph/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+	RegisterSignal(src, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, PROC_REF(pre_attack))
+	RegisterSignal(src, COMSIG_CLICK_SHIFT, PROC_REF(trigger_ability))
+	RegisterSignal(src, COMSIG_ACTION_DISGUISED_APPEARANCE, PROC_REF(on_disguise))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_DISGUISED), PROC_REF(on_undisguise))
+
+	AddElement(/datum/element/content_barfer)
+
+	disguise_ability = new(src)
+	disguise_ability.Grant(src)
+
+/mob/living/basic/morph/examine(mob/user)
+	if(!HAS_TRAIT(src, TRAIT_DISGUISED))
+		return ..()
+
+	var/atom/movable/form_reference = form_weakref.resolve()
+	if(!isnull(form_reference))
+		. = form_reference.examine(user)
+
+	if(get_dist(user, src) <= 3) // always add this because if the form_reference somehow nulls out we still want to have something look "weird" about an item when someone is close
+		. += span_warning("It doesn't look quite right...")
+
+/mob/living/basic/morph/med_hud_set_health()
+	if(isliving(form_typepath))
+		return ..()
+
+	//we hide medical hud while in regular state or an item
+	var/image/holder = hud_list[HEALTH_HUD]
+	holder.icon_state = null
+
+/mob/living/basic/morph/med_hud_set_status()
+	if(isliving(form_typepath))
+		return ..()
+
+	//we hide medical hud while in regular state or an item
+	var/image/holder = hud_list[STATUS_HUD]
+	holder.icon_state = null
+
+/mob/living/basic/morph/death(gibbed)
+	if(!HAS_TRAIT(src, TRAIT_DISGUISED))
+		return ..()
+
+	// the action should handle the rest
+	visible_message(
+		span_warning("[src] twists and dissolves into a pile of green flesh!"),
+		span_userdanger("Your skin ruptures! Your flesh breaks apart! No disguise can ward off de--"),
+	)
+
+	return ..()
```

Comment:
```suggestion
	if(HAS_TRAIT(src, TRAIT_DISGUISED))
		// the action should handle the rest
		visible_message(
			span_warning("[src] twists and dissolves into a pile of green flesh!"),
			span_userdanger("Your skin ruptures! Your flesh breaks apart! No disguise can ward off de--"),
		)

	return ..()
```
As much as I am early-return pilled are we really saving anything here

Reply:
the indentation looks less ugly